api/lib/utils.js

```diff
@@ -115,6 +117,34 @@ function isValidLedgerWord(ledger) {
   return (/^current$|^closed$|^validated$/.test(ledger));
 }
 
+function attachDate(api, baseTransactions, callback) {
+  var groupedTx = _.groupBy(baseTransactions, function(tx) {
+    return tx.ledger_index;
+  });
+
+  async.each(_.keys(groupedTx), function(ledger, next) {
```

Comment:
>  I can read this fine
At it's deepest, the paren+curly depth is 7, which is pretty smelly. I think it's worth changing because if we don't do it now it will probably get ignored for a long time.